test/Garnet.test/RespCustomCommandTests.cs

```diff
@@ -10,13 +10,62 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Garnet.common;
 using Garnet.server;
 using NUnit.Framework;
 using NUnit.Framework.Legacy;
 using StackExchange.Redis;
+using Tsavorite.core;
 
 namespace Garnet.test
 {
+    public class LargeGet : CustomProcedure
+    {
+        public override bool Execute(IGarnetApi garnetApi, ArgSlice input, ref MemoryResult<byte> output)
+        {
+            int offset = 0;
+            var key = GetNextArg(input, ref offset);
+            for (int i = 0; i < 120_000; i++)
+            {
+                garnetApi.GET(key, out var outval);
+                garnetApi.FreeBuffer(ref outval);
+            }
+
+            garnetApi.GET(key, out var outval1);
+            garnetApi.GET(key, out var outval2);
+            garnetApi.FreeBuffer(ref outval2);
+            garnetApi.FreeBuffer(ref outval1); // Ensure graceful handling of earlier allocated buffer
```

Comment:
test what happens if they first free 1 and then 2

Reply:
That is getting handled gracefully as the offsets and length check within the scratchbuffer manager will fail. Added a test to verify that.